### DISCUSSION TITLE
How does competing dashboards blocking work?

### DISCUSSION THREAD

**Discussion by bahmutov** [cypress-io/cypress#28138] [Plugins]
In the blog post "Changes in defense of Cypress Intellectual Property" https://www.cypress.io/blog/2023/09/25/defense-intellectual-property you have announced blocking competing dashboards. Looking at the plugins that are blocked (https://currents.dev/posts/v13-blocking, https://github.com/agoldis/cypress-blocking-example, https://news.ycombinator.com/item?id=37842778) with names like:

- @deploysentinel/cypress-debugger
- @deploysentinel/cypress-parallel
- cypress-debugger
- @currents/cypress-debugger-support
- @currents/cypress-debugger-plugin
- cypress-cloud
- cypress-debug
- cypress-vscode

We have a question: how are these NPM modules discovered? Where is the code to actually find them and show this error message?

```
We've detected that you're using a 3rd party library that is not supported by Cypress: cypress-cloud
To continue running Cypress, please remove this library or reach out for help migrating.
https://on.cypress.io/unsupported-third-party-library
```

I searched the Cypress main repo for any strings above and could not find it. Is the code to detect and block the execution coming from another repository? Is it injected into the binary at the build step? Is the detection and blocking code covered by the MIT license?

**Comment by estefafdez** [comment]
+1 to all those questions

**Comment by vaclavGabriel** [comment]
Can we get some answers, please? @jennifer-shehane

**Comment by firmanserdana** [comment]
Bump, for the sake of open source-ness, even one of those 3rd party is, in the end, open source.
